[Pages] Add improvements and troubleshooting guide to git-integration.md (#11598)
cloudflare/cloudflare-docs: content/pages/platform/git-integration.md

![Pausing a deployment in the Settings of your Pages project](/images/pages/platform/git.pause.png)

Selecting **Pause deployments** will present a confirmation and, once confirmed, the **Pause deployments** button will be replaced with a **Resume deployments** button. While paused, your **Deployments** list will present a banner message, reminding you that automatic deployments are not enabled.

> ## Reinstalling a git installation

Reviewer: - [ ] ... `a Git installation`

> When encountering Git integration related issues, one potential troubleshooting step is attempting to uninstall and reinstall an installation. The process for each git provider is provided below

Reviewer: technically you're un/re-installing the app not the installation

Reviewer: - [ ] `The process for each Git provider`...

### GitHub

1. Go to the installation settings page on GitHub
    1. `https://github.com/settings/installations` for individual accounts
    1. `https://github.com/organizations/[YOUR_ORGANIZATION_NAME]/settings/installations` for organizational accounts

> 2. If the Cloudflare Pages installation is there, click `Configure`, and click `Uninstall 'Cloudflare Pages'` (if there is no "Cloudflare Pages" installation, the user doesn't need to do anything)

Reviewer: we should use the same quote convention when spelling Cloudflare Pages, meaning either both with "" or both without

> 3. Go back to the `Workers & Pages` overview page at `https://dash.cloudflare.com/[YOUR\_ACCOUNT\_ID]/workers-and-pages`. Click `Create application` > `Pages` > `Connect to git`

Reviewer: - [ ] `Connect to Git`

4. Click the '+ Add account' button, click the GitHub account you want to add, and then click `Install & Authorize`.
5. You should be redirected to the create project page with your GitHub account or organization in the account list.
6. Attempt to make a new deployment with your project which was previously broken.

### GitLab

1. Go to your application settings page on GitLab located here: https://gitlab.com/-/profile/applications
2. Click the "Revoke" button on your Cloudflare Pages installation if it exists.

> 3. Go back to the `Workers & Pages` overview page at `https://dash.cloudflare.com/[YOUR\_ACCOUNT\_ID]/workers-and-pages`. Click `Create application` > `Pages` > `Connect to git`

Reviewer: - [ ] `Connect to Git`

4. Select the GitLab tab at the top, click the '+ Add account' button, click the GitLab account you want to add, and then click `Authorize` on the modal titled "Authorize Cloudflare Pages to use your account? ".

> 5. You should be redirected to the create project page with your GitHub account or organization in the account list.

Reviewer: - [ ] `GitLab` not `GitHub`

6. Attempt to make a new deployment with your project which was previously broken.

## Troubleshooting

> If you run into any issues related to deployments failing, check your project dashboard to see if there are any SCM installation warnings listed. To resolve any errors displayed in the Cloudflare Pages dashboard, follow the steps listed below.

Reviewer: should we add a screenshot here with an example of such a warning?

Author: - [ ] will add

> ### `The Cloudflare Pages installation was uninstalled on GitHub. To reinstall, log in to the Cloudflare dashboard > Workers & Pages > Create an application > Connect to git > + Add account.`

Reviewer: - [ ] `Connect to Git`

To resolve this issue, please follow the steps provided above in the [Reinstalling a git installation section](/pages/platform/git-integration/#reinstalling-a-git-installation) for the applicable SCM provider. If the issue persists even after uninstalling and reinstalling, please contact support.

> ### `Cloudflare Pages is not properly installed on your git account. Please uninstall and reinstall your git installation to make new deployments.`

Reviewer: - [ ] `Git account`

- [ ] `Git installation`

To resolve this issue, please follow the steps provided above in the [Reinstalling a git installation section](/pages/platform/git-integration/#reinstalling-a-git-installation) for the applicable SCM provider. If the issue persists even after uninstalling and reinstalling, please contact support.

### `The Cloudflare Pages installation has been suspended. Please unsuspend in your GitHub installation settings.`

Go to your GitHub installation settings located at:

* `https://github.com/settings/installations` for individual accounts
* `https://github.com/organizations/[YOUR_ORGANIZATION_NAME]/settings/installations` for organizational accounts

Click "Configure" on the Cloudflare Pages application. Scroll down to the bottom of the page and click "Unsuspend" to allow Cloudflare Pages to make future deployments.

> ### `You have no linked repositories matching the provided name. Please verify this repository exists on the correct git account.`

Reviewer: - [ ] `Git account`

You may have deleted or transferred the repository associated with this Cloudflare Pages project. For a deleted repository, you will need to create a new Cloudflare Pages project with a repository that has not been deleted. For a transferred repository, you can either transfer the repository back to the original Git account or you will need to create a new Cloudflare Pages project with the transferred repository.

### `This repo is unable to be accessed. Please review your GitHub repo access permissions for the Cloudflare Pages installation to make new deployments.`

You may have excluded this repository from your installation's repository access settings.  Go to your GitHub installation settings located at:

* `https://github.com/settings/installations` for individual accounts
* `https://github.com/organizations/[YOUR_ORGANIZATION_NAME]/settings/installations` for organizational accounts

Click "Configure" on the Cloudflare Pages application. Under "Repository access," ensure that the repository associated with your Cloudflare Pages project is included in the list.

> ### `There is an issue with your Cloudflare Pages git installation. Please contact support.`

Reviewer: - [ ] `Git installation`

> This is an internal error in the Cloudflare Pages SCM system. You can attempt to [reinstalling your git installation](/pages/platform/git-integration/#reinstalling-a-git-installation), but if the issue persists please contact support.

Reviewer: You can attempt to `reinstall` ...

6. Attempt to make a new deployment with your project which was previously broken.

### GitLab

1. Go to your application settings page on GitLab located here: https://gitlab.com/-/profile/applications
2. Click the "Revoke" button on your Cloudflare Pages installation if it exists.
3. Go back to the `Workers & Pages` overview page at `https://dash.cloudflare.com/[YOUR\_ACCOUNT\_ID]/workers-and-pages`. Click `Create application` > `Pages` > `Connect to Git`
4. Select the GitLab tab at the top, click the '+ Add account' button, click the GitLab account you want to add, and then click `Authorize` on the modal titled "Authorize Cloudflare Pages to use your account? ".
5. You should be redirected to the create project page with your GitHub account or organization in the account list.
6. Attempt to make a new deployment with your project which was previously broken.

## Troubleshooting

If you run into any issues related to deployments failing, check your project dashboard to see if there are any SCM installation warnings listed as shown in the screenshot below.

![Pausing a deployment in the Settings of your Pages project](/images/pages/platform/git.dashboard-error.png)

> To resolve any errors displayed in the Cloudflare Pages dashboard, follow the steps listed below.

Reviewer 2: ```suggestion
To resolve any errors displayed in the Cloudflare Pages dashboard, refer to the following instructions.
```

### `This project is disconnected from your Git account, this may cause deployments to fail.`

> To resolve this issue, please follow the steps provided above in the [Reinstalling a Git installation section](/pages/platform/git-integration/#reinstalling-a-git-installation) for the applicable SCM provider. If the issue persists even after uninstalling and reinstalling, please contact support.

Reviewer 2: ```suggestion
To resolve this issue, follow the steps provided above in the [Reinstalling a Git installation section](/pages/platform/git-integration/#reinstalling-a-git-installation) for the applicable SCM provider. If the issue persists even after uninstalling and reinstalling, contact support.
```

### `Cloudflare Pages is not properly installed on your Git account, this may cause deployments to fail.`

> To resolve this issue, please follow the steps provided above in the [Reinstalling a Git installation section](/pages/platform/git-integration/#reinstalling-a-git-installation) for the applicable SCM provider. If the issue persists even after uninstalling and reinstalling, please contact support.

Reviewer 2: ```suggestion
To resolve this issue, follow the steps provided above in the [Reinstalling a Git installation section](/pages/platform/git-integration/#reinstalling-a-git-installation) for the applicable SCM provider. If the issue persists even after uninstalling and reinstalling, contact support.
```

### `The Cloudflare Pages installation has been suspended, this may cause deployments to fail.`

Go to your GitHub installation settings located at:

> * `https://github.com/settings/installations` for individual accounts
> * `https://github.com/organizations/[YOUR_ORGANIZATION_NAME]/settings/installations` for organizational accounts
> 
> Click "Configure" on the Cloudflare Pages application. Scroll down to the bottom of the page and click "Unsuspend" to allow Cloudflare Pages to make future deployments.

Reviewer 2: ```suggestion
* `https://github.com/settings/installations` for individual accounts.
* `https://github.com/organizations/[YOUR_ORGANIZATION_NAME]/settings/installations` for organizational accounts.

Select **Configure** on the Cloudflare Pages application. Scroll down to the bottom of the page and select **Unsuspend** to allow Cloudflare Pages to make future deployments.
```

### `The project is linked to a repository that no longer exists, this may cause deployments to fail.`

You may have deleted or transferred the repository associated with this Cloudflare Pages project. For a deleted repository, you will need to create a new Cloudflare Pages project with a repository that has not been deleted. For a transferred repository, you can either transfer the repository back to the original Git account or you will need to create a new Cloudflare Pages project with the transferred repository.

### `The repository is unable to be accessed, this may cause deployments to fail.`

> You may have excluded this repository from your installation's repository access settings.  Go to your GitHub installation settings located at:

Reviewer 2: ```suggestion
You may have excluded this repository from your installation's repository access settings.  Go to your GitHub installation settings:
```

> * `https://github.com/settings/installations` for individual accounts
> * `https://github.com/organizations/[YOUR_ORGANIZATION_NAME]/settings/installations` for organizational accounts

Reviewer 2: ```suggestion
* `https://github.com/settings/installations` for individual accounts.
* `https://github.com/organizations/<YOUR_ORGANIZATION_NAME>/settings/installations` for organizational accounts.
```

> Click "Configure" on the Cloudflare Pages application. Under "Repository access," ensure that the repository associated with your Cloudflare Pages project is included in the list.

Reviewer 2: ```suggestion
Select **Configure** on the Cloudflare Pages application. Under **Repository access**, ensure that the repository associated with your Cloudflare Pages project is included in the list.
```
